### ISSUE TITLE
Enable CA2000 across project?

### ISSUE THREAD

**Issue by lsoft** [AvaloniaUI/Avalonia#13457]
There was errors with disposable objects like https://github.com/AvaloniaUI/Avalonia/issues/5294.

Please consider enable and using CA2000 across project, at least once. The main difficulty is CA2000 work on net7 project, but does not work on netstandard2.0 projects.

I manually switched `Avalonia.Build.Tasks.csproj` to net7 (and broke it, of course), add `dotnet_diagnostic.CA2000.severity = error` to  `.editorconfig`, removed `<EnableNETAnalyzers>false</EnableNETAnalyzers>` node from the `csproj` and tried to build `Avalonia.Build.Tasks.csproj`.

Additional catches are found:

```
21:51:13:421	1>C:\projects\Avalonia\src\Avalonia.Build.Tasks\XamlCompilerTaskExecutor.Helpers.cs(44,30,44,54): error CA2000: Call System.IDisposable.Dispose on object created by 'new BinaryReader(stream)' before all references to it are out of scope (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

21:51:13:443	1>C:\projects\Avalonia\src\Markup\Avalonia.Markup.Xaml.Loader\CompilerExtensions\Transformers\AvaloniaXamlIlDeferredResourceTransformer.cs(85,31,85,73): error CA2000: Call System.IDisposable.Dispose on object created by 'emitter.LocalsPool.GetLocal(Parameters[c])' before all references to it are out of scope (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

21:51:13:443	1>C:\projects\Avalonia\src\Avalonia.Build.Tasks\XamlCompilerTaskExecutor.cs(285,36,285,88): error CA2000: Call System.IDisposable.Dispose on object created by 'new StreamReader(new MemoryStream(res.FileContents))' before all references to it are out of scope (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

```

probably it is a minor bugs, but I did not check a whole avalonia repo.

I would suggest to set all csproject to net7, enable the analyzer, fix all errors and return `TargetFramework`s to the default ones.

Thanks for your work!

**Comment by maxkatz6** [comment]
CA2000 should work on .NET 6 projects, which is enough for us, as most of projects are netstandard2.0;net6.0 in our repo.
I would say it's a subtask of another issue: https://github.com/AvaloniaUI/Avalonia/issues/8944

**Comment by maxkatz6** [comment]
The reason why Avalonia.Build.Tasks targets only netstandard2.0 is that it's shipped with nuget package as a tool and is not directly consumed by the user. Instead, MSBuild (either netframework or netcore) runs it as a netstandard2.0. 
But we can at least add a second target only for warnings and errors, while continuing to use only netstandard version.

**Comment by lsoft** [comment]
> The reason why Avalonia.Build.Tasks targets only netstandard2.0

Yeah, I know it. I only suggested switch the csproj to net7 temporarily and locally, find all issues, fix it and switch target framework back to netstandard2.0, and then push the changes to repo.

> I would say it's a subtask of another issue

that issue does not contains CA2000. should I close this issue anyway?

**Comment by workgroupengineering** [comment]
I have update issue #8944

**Comment by lsoft** [comment]
@workgroupengineering thanks!
